### PR TITLE
Allow only keeping a subset of slides for SlideSet

### PR DESCRIPTION
**Pull request by kheffah**
## Description

Allow only keeping a subset of slides for SlideSet. This would be useful when, for example, you want to exclude slides used in machine learning model training for cross validation.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [x] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Fixes: 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/histolab/histolab/blob/master/CONTRIBUTING.md) document.
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: src/histolab/slide.py [hunk 1]
```diff
@@ -444,11 +444,16 @@ class SlideSet:
     """Slideset object. It is considered a collection of Slides."""
 
     def __init__(
-        self, slides_path: str, processed_path: str, valid_extensions: list
+        self,
+        slides_path: str,
+        processed_path: str,
+        valid_extensions: list,
```
Comment (alessiamarcolini) [inline src/histolab/slide.py:450]:
```suggestion
        valid_extensions: List[str],
```

File: src/histolab/slide.py [hunk 2]
```diff
@@ -444,11 +444,16 @@ class SlideSet:
...
+        keep_slides: list = None,
```
Comment (alessiamarcolini) [inline src/histolab/slide.py:451]:
```suggestion
        keep_slides: List[str] = None,
```

File: tests/unit/test_slide.py [hunk 3]
```diff
@@ -384,15 +384,22 @@ def it_constructs_from_args(self, request):
         _slides_path = "/foo/bar/"
         _processed_path = "/foo/bar/wsislides/processed"
         _valid_extensions = [".svs", ".tiff"]
+        _keep_slides = ["mywsi.svs"]
 
-        slideset = SlideSet(_slides_path, _processed_path, _valid_extensions)
+        slideset = SlideSet(
+            _slides_path, _processed_path, _valid_extensions, keep_slides=_keep_slides
```
Comment (alessiamarcolini) [inline tests/unit/test_slide.py:390]:
```suggestion
            _slides_path, _processed_path, _valid_extensions, _keep_slides
```
I would stick for the same arguments style (positional) for all of them ;)

File: tests/unit/test_slide.py [hunk 4]
```diff
@@ -384,15 +384,22 @@ def it_constructs_from_args(self, request):
...
+        )
 
         _init_.assert_called_once_with(
-            ANY, _slides_path, _processed_path, _valid_extensions
+            ANY,
+            _slides_path,
+            _processed_path,
+            _valid_extensions,
+            keep_slides=_keep_slides,
```
Comment (alessiamarcolini) [inline tests/unit/test_slide.py:398]:
```suggestion
            _keep_slides,
```

**Review by alessiamarcolini — CHANGES_REQUESTED** [review]
Just a couple of stylistic changes :) the rest is good, thanks!!

**Comment by kheffah** [comment]
@alessiamarcolini Thanks, all done 👍🏻

File: src/histolab/slide.py [hunk 5]
```diff
@@ -457,11 +462,17 @@ def __iter__(self) -> Iterator[Slide]:
         -------
         generator of `Slide` objects.
         """
+        slide_names = [
+            _name
```
Comment (ernestoarbitrio) [inline src/histolab/slide.py:466]: I don't understand the need of the underscore before the name. I'd use name without any special characters.

**Review by ernestoarbitrio — CHANGES_REQUESTED** [review]
Per comments

File: src/histolab/slide.py (hunk 5 above)
Comment (kheffah) [reply src/histolab/slide.py]: @ernestoarbitrio I agree. I just retained the variable name from the original version of the file. OK, removed the underscore.

**Review by ernestoarbitrio — APPROVED** [review]
LGTM

**Review by alessiamarcolini — APPROVED** [review]
LGTM!! 

However we need to wait to merge this PR as it is scheduled to be in the 0.2.8 release :)

**Comment by alessiamarcolini** [comment]
@kheffah if you don't mind, I'll rebase on master and squash the "update" commits

**Comment by kheffah** [comment]
> @kheffah if you don't mind, I'll rebase on master and squash the "update" commits

Sure thing, feel free :)

**Pull request merged**